packages/react-components/src/utils/icon.ts

```diff
@@ -0,0 +1,31 @@
+import { EmotionJSX } from '@emotion/react/types/jsx-namespace';
+
+import {
+  protocol,
+  article,
+  dataset,
+  bioinformatics,
+  labResource,
+  grantDocument,
+} from '../icons';
+
+export const getIconForDocumentType = (
+  documentType: string,
+): EmotionJSX.Element => {
+  switch (documentType) {
+    case 'Protocol':
+      return protocol;
+    case 'Article':
+      return article;
+    case 'Dataset':
+      return dataset;
+    case 'Bioinformatics':
+      return bioinformatics;
+    case 'Lab Resource':
+      return labResource;
+    case 'Grant Document':
+      return grantDocument;
+    default:
+      return protocol;
+  }
+};
```

Comment:
```
const icons: Record<ResearchOutputDocumentType, EmotionJSX.Element> = {
  Protocol: protocol,
  Article: article,
  Dataset: dataset,
  Bioinformatics: bioinformatics,
  'Lab Resource': labResource,
  'Grant Document': grantDocument,
  Presentation: protocol,
  Report: protocol,
};

export const getIconForDocumentType = (
  documentType: ResearchOutputDocumentType,
): EmotionJSX.Element => icons[documentType];
```

Reply:
This we refactors from this lookup, (which I prefer to a switch statement).
But also has 2 extra properties: Presentation and Report.

Reply:
committed here [review: refactor getIconForDocumentType](https://github.com/yldio/asap-hub/pull/3816/commits/7a42a73e73570b5d4122a09a1c6649f9c57d30d1)

I had some Typescript issues though. When passing a default value for `getIconForDocumentType` like 

```
const RelatedResearchCard: React.FC<RelatedResearchCardProp> = ({
  relatedResearch,
  description,
  getSourceIcon,
  getIconForDocumentType = getIconForDocumentTypeCRN,
  title = 'Related Research',
  tableTitles = [
    'Document Type',
    'Shared Output Name',
    'Team or Working Group',
  ],
}) => {
```

Typescript somehow couldn't accept that `getIconForDocumentType` could receive a gp2.OutputDocumentType documentType so I kept getting this error

```
Type '"Code/Software"' is not assignable to type '"Grant Document" | "Presentation" | "Dataset" | "Bioinformatics" | "Protocol" | "Lab Resource" | "Article" | "Report"'
```

Then I removed the default value and typed `getIconForDocumentType` as 

```
getIconForDocumentType:
  | ((documentType: ResearchOutputDocumentType) => EmotionJSX.Element)
  | ((documentType: gp2.OutputDocumentType) => EmotionJSX.Element);
```

but then Typescript gives this error

```
 Argument of type '"Dataset" | "Article" | "Code/Software" | "GP2 Reports" | "Procedural Form" | "Training Materials"' is not assignable to parameter of type '"Dataset" | "Article"'.
```

when I try to call `getIconForDocumentType` as if it only accepts the intersection of the union types. So the only way I've found to not give an error was to call getIconForDocumentType casting documentType as any. This is a shame, but I haven't found another way to do it. I'm very pleased to accept suggestions though.